src/runtime_src/xdp/profile/rt_profile_writers.cpp

```diff
@@ -1095,14 +1098,23 @@ namespace XCL {
     if(SummaryFileName != "") {
       assert(!Summary_ofs.is_open());
       SummaryFileName += FileExtension;
-      openStream(Summary_ofs, SummaryFileName);
+      try {
+        openStream(Summary_ofs, SummaryFileName);
+      } catch(std::runtime_error &err) {
+        std::cout << " ERROR : " << err.what() << " SDAccel Profile Summary." << std::endl;
+      }
       writeDocumentHeader(Summary_ofs, "SDAccel Profile Summary");
     }
 
     if (TimelineFileName != "") {
       assert(!Timeline_ofs.is_open());
       TimelineFileName += FileExtension;
-      openStream(Timeline_ofs, TimelineFileName);
+      try {
+        openStream(Timeline_ofs, TimelineFileName);
+      } catch(std::runtime_error &err) {
+        std::cout << " ERROR : " << err.what() << " SDAccel Timeline Trace." << std::endl;
+      }
```

Comment:
Is it really safe to continue to writeDocumentHeader when an exception was thrown?

Reply:
Hi Soren,
writeDocumentHeader itself checks whether the passed ofstream is open. WriterI::openStream throws the exception when the ofstream could not be opened. So, handling the exception and generating error message should be sufficient. All the other methods using the ofsteam checks whether it is open.
Thanks!

Reply:
Ok, but this seems a bit odd.   Catching is supposed to handle the exception, but it really doesn't, why not just rethrow?